src/Network/Request.php

```diff
@@ -1172,6 +1172,94 @@ public function param($name)
         }
         return $this->params[$name];
     }
+    
+    /**
+	 * Extracts values from $this->data or $this->query and returns them (by default)
+	 * as a prepared hash with dot notation as a key and value as the value to be used
+	 * for instance for feeding the ORM.
+	 *
+	 * The second parameter $options configures the extraction:
+	 * 1. The unnamed option [0] may be set to 'query' or 'data', and defaults to query,
+	 *    where 'query' will try to extract from the request query string and 'data'
+	 *    will try to extract from the request data array.
+	 *
+	 * 2. By default this method will return a hash with dot notation prefix as the key,
+	 *    and the extracted value as a value, as convenience to be used by the query builder.
+	 *    The dotnotation returned will only return one level up, e.g. extracting
+	 *    Users.Profiles.is_visible will return given value on that path with the key
+	 *    'Profiles.is_visible'. to be able to inject into ORM's matching() and alike.
```

Comment:
I had my issues as well, but the goal was to implement some extractor that can be passed to where(), matching() etc. calls. And there only Modelname.fieldname is expected.
Because I had my issues I did add a fullPath option.